### PR TITLE
Add `H` as an alias for Hadamard operator

### PR DESCRIPTION
**Pull request by Alex-Preciado**
This PR adds `H` as an alias for the `Hadamard` operator. Main goal is to enhance usability and consistency in gate naming.

**Related Shortcut Stories:**
[sc-70976]

### CODE REVIEW

File: pennylane/ops/qubit/non_parametric_ops.py [hunk 1]
```diff
@@ -182,6 +195,24 @@ def pow(self, z: Union[int, float]):
         return super().pow(z % 2)
 
 
+H = Hadamard
+r"""Hadamard(wires)
```
Comment (albi3ro) [inline pennylane/ops/qubit/non_parametric_ops.py:199]:
```suggestion
r"""H(wires)
```

File: tests/ops/qubit/test_non_parametric_ops.py
Comment (albi3ro) [inline tests/ops/qubit/test_non_parametric_ops.py:1]: Might be worth just adding a short test in this file making sure that `assert qml.H is qml.Hadamard`.  And maybe `assert isinstance(qml.H(0), qml.Hadamard)` or something like that.

File: doc/releases/changelog-dev.md [hunk 2]
```diff
@@ -189,6 +189,9 @@
 
 <h4>Other Improvements</h4>
 
+* Added `H` as an alias for the Hadamard operator.
```
Comment (albi3ro) [inline doc/releases/changelog-dev.md:192]:
```suggestion
* Added `qml.H` as an alias for the Hadamard operator.
```

**Review by albi3ro — APPROVED** [review]
Not entirely sure if we need to add it to the device test suite, since that's more about making sure devices support the various operations, but can't hurt.

This PR will definitely make a lot of people happy.

File: tests/ops/qubit/test_non_parametric_ops.py
Comment (Alex-Preciado) [reply tests/ops/qubit/test_non_parametric_ops.py:1]: Good idea! Adding them now 🚀

Comment (Alex-Preciado) [reply tests/ops/qubit/test_non_parametric_ops.py:1]: Just added some tests to `test_non_parametric_ops.py`

**Review by mudit2812 — APPROVED** [review]

File: tests/ops/qubit/test_non_parametric_ops.py
Comment (Alex-Preciado) [reply tests/ops/qubit/test_non_parametric_ops.py:1]: Requested tests have been added. Resolving this conversation now. Many thanks @albi3ro 🙏🏼

**Pull request merged**